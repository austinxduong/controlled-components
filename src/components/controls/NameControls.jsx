import React from 'react';
import PropTypes from 'prop-types';


// eslint-disable-next-line react/prop-types
const NameControls = ({ greeting, name, onChange }) => (
  <>
    <label htmlFor="greeting"> Greeting </label>
    <input
      id="greeting"
      type="greeting"
      placeholder="Greeting Message"
      name="greeting"
      value={greeting}
      onChange={onChange}
    />
  
    <label htmlFor="name"> Name </label>
    <input
      id="name"
      type="name"
      placeholder="Name"
      name="name"
      value={name}
      onChange={onChange}
    />

  </>
);

NameControls.PropTypes = { 
  greeting: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default NameControls;