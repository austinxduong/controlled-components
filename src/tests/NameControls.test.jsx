import React from 'react';
import { render } from '@testing-library/react';
import NameControls from '../components/controls/NameControls';

describe('test functionality for NameControls Component', () => {
  it('renders the name input to display', () => {
    const { asFragment } = render (
      <NameControls text=""
      />
    );
    expect(asFragment()).toMatchSnapshot();
  }
  );
}
);
