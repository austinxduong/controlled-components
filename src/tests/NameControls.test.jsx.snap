// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameControls component NameControls 1`] = `
<DocumentFragment>
  <label
    for="greeting
  >
    Greeting
  </label>
  <input
    id="greeting"
    name="greeting"
    value=""
  />
  <label
    for="name"
  >
    Name
  </label>
  <input
    id="name"
    name="name"
    placeholder="Name"
    type="name"
    value=""
  />
</DocumentFragment>
`;
